packages/zapp/console/src/components/Executions/Tables/NodeExecutionsTable.tsx

```diff
@@ -26,17 +31,45 @@ const scrollbarPadding = scrollbarSize();
  * NodeExecutions are expandable and will potentially render a list of child
  * TaskExecutions
  */
-export const NodeExecutionsTable: React.FC<NodeExecutionsTableProps> = ({
+export const NodeExecutionsTable: FC<NodeExecutionsTableProps> = ({
   abortMetadata,
-  nodeExecutions,
+  initializeNodes,
 }) => {
-  const [selectedExecution, setSelectedExecution] = React.useState<NodeExecutionIdentifier | null>(
-    null,
-  );
+  const [selectedExecution, setSelectedExecution] = useState<NodeExecutionIdentifier | null>(null);
+  const [nodeExecutions, setNodeExecutions] = useState<NodeExecution[]>([]);
   const commonStyles = useCommonStyles();
   const tableStyles = useExecutionTableStyles();
+  const nodeExecutionsById = useContext(NodeExecutionsByIdContext);
+
+  useEffect(() => {
+    if (nodeExecutionsById) {
+      const executions: NodeExecution[] = [];
+      initializeNodes.map((node) => {
+        if (nodeExecutionsById[node.scopedId]) executions.push(nodeExecutionsById[node.scopedId]);
+        else
+          executions.push({
+            closure: {
+              createdAt: dateToTimestamp(new Date()),
+              outputUri: '',
+              phase: NodeExecutionPhase.UNDEFINED,
+            },
+            id: {
+              executionId: {
+                domain: node.value?.taskNode?.referenceId?.domain,
+                name: node.value?.taskNode?.referenceId?.name,
+                project: node.value?.taskNode?.referenceId?.project,
+              },
+              nodeId: node.id,
+            },
+            inputUri: '',
+            scopedId: node.scopedId,
+          });
+      });
+      setNodeExecutions(executions);
+    }
+  }, [nodeExecutionsById, initializeNodes]);
```

Comment:
if `initializeNodes` changes, whole component re-renders, as it's a prop, so no need to rely on it in the dependency

Reply:
I don't think so